File: TypeScript/src/bruteforce.ts
const ALPHANUMERIC: string = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz1234567890";

function getGuessLength(minLength: number, maxLength: number): number {
  const length: number = Math.floor(Math.random() * maxLength);

    if( length < minLength ) {
        return minLength;
    }

    return length;
}

function getRandomCharacter(): string {
  const randomPosition = Math.floor(Math.random() * ALPHANUMERIC.length);

  return ALPHANUMERIC.charAt(randomPosition);
}

export default function bruteforcePassword(password: string, minLength: number,  maxLength: number) : number {  
  let isNotCracked: boolean = true;
  let guesses: number = 1;

  while(isNotCracked) {
      const length = getGuessLength(minLength, maxLength);
      let currentGuess = '';

      for(let i = 0; i < length; i += 1){
        currentGuess = currentGuess.concat(getRandomCharacter());
      }
      console.log(`${guesses} guess is: ${currentGuess}`);

      if(password === currentGuess) {
          console.log('Found a match!!');
          console.log(currentGuess);
          isNotCracked = false;
      }

      guesses++;
  }

  return guesses;
} 

File: Python/src/bruteforce.py
from random import randrange

def get_random_character() -> str:
    alphanumerical = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz1234567890"
    random_position = randrange(len(alphanumerical))

    return alphanumerical[random_position]

def bruteforce_password(password: str, min_length: int, max_length: int) -> int:
    is_not_cracked = True
    guesses = 1

    while is_not_cracked:
        length = randrange(min_length, max_length)
        current_guess = ""

        for i in range(0, length):
            current_guess += get_random_character()
        
        print(guesses, "guess is:", current_guess)

        if current_guess == password:
            print("Found a match!")
            print(current_guess)
            is_not_cracked = False
        
        guesses += 1
    
    return guesses